lib/stistools/inttag.py

```diff
@@ -52,181 +100,176 @@ def inttag(tagfile, output, starttime=None, increment=None,
-        gti_data = np.rec.array([(events_data['TIME'][0], events_data['TIME'][-1])],
-                                formats=">f8,>f8", names='START, STOP')
-    else:  # Otherwise, retrieve the GTIs from the GTI extension
-        gti_data = tag_hdr['GTI'].data
-
-        # C code checked if it could retrieve a GTI table,
-        # If not, it used the events table. Not sure if we
-        # want to allow this, leaving it out for now
-
-    # Determine start and stop times for counting events
-    gti_start = gti_data['START'][0]
-    gti_stop = gti_data['STOP'][-1]
-
-    # Get Header Info
-    cenx = tag_hdr[0].header['CENTERA1']  # xcenter in c code
-    ceny = tag_hdr[0].header['CENTERA2']  # ycenter in c code
-    siz_axx = tag_hdr[0].header['SIZAXIS1']  # nx in c code
-    siz_axy = tag_hdr[0].header['SIZAXIS2']  # ny in c code
-    tzero_mjd = tag_hdr[1].header['EXPSTART']  # MJD zero point
-
-    xcorner = ((cenx - siz_axx / 2.) - 1) * 2
-    ycorner = ((ceny - siz_axy / 2.) - 1) * 2
-
-    # Adjust axis sizes for highres, determine binning
-    bin_n = 2
-    if highres:
-        siz_axx *= 2
-        siz_axy *= 2
-        bin_n = 1
-
-    ltvx = ((bin_n - 2.) / 2. - xcorner) / bin_n
-    ltvy = ((bin_n - 2.) / 2. - ycorner) / bin_n
-    ltm = 2. / bin_n
-
-    # Read in start and stop time parameters
-    if starttime is None:
-        starttime = gti_start  # The first START time in the GTI (or first event)
-
-    if increment is None:
-        increment = (gti_stop - gti_start)/rcount
-    stoptime = starttime + increment
-
-    imset_hdr_ver = 0  # output header value corresponding to imset
-    texptime = 0  # total exposure time
-    hdu_list = []
-    for imset in range(rcount):
-
-        # Get Exposure Times
-        exp_time, expstart, expstop, good_events = exp_range(starttime, stoptime, events_data, gti_data, tzero_mjd)
-        if len(good_events) == 0:
+    with fits.open(tagfile) as tag_hdr:
+        # Read in TIME-TAG data (Events data and GTI data)
+        events_data = tag_hdr[1].data
+        if allevents:  # If allevents, ignore GTI and generate gti_data based on the time of the first and last event
+            gti_data = np.rec.array([(events_data['TIME'][0], events_data['TIME'][-1])],
+                                    formats=">f8,>f8", names='START, STOP')
+        else:  # Otherwise, retrieve the GTIs from the GTI extension
+            gti_data = tag_hdr['GTI'].data
+
+        # Read in relevant header info
+        tag_pri_hdr = tag_hdr[0].header
+        cenx = tag_pri_hdr['CENTERA1']  # xcenter in c code
+        ceny = tag_pri_hdr['CENTERA2']  # ycenter in c code
+        siz_axx = tag_pri_hdr['SIZAXIS1']  # nx in c code
+        siz_axy = tag_pri_hdr['SIZAXIS2']  # ny in c code
+        tag_sci_hdr = tag_hdr[1].header
+        tzero_mjd = tag_sci_hdr['EXPSTART']  # MJD zero point
+
+        # Determine start and stop times for counting events
```

Comment:
It looks like you can un-indent from the `with` section starting here since you no longer use `tag_hdr`. Other then that 👍

Reply:
Ah okay great. Fixed it, thanks!